src/lib/comms/sol-lwm2m.c

```diff
@@ -3085,6 +3163,127 @@ handle_create(struct sol_lwm2m_client *client,
     return SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
 }
 
+static uint8_t
+handle_write(struct sol_lwm2m_client *client,
+    struct obj_ctx *obj_ctx, struct obj_instance *obj_instance,
+    uint16_t *path, uint16_t path_size,
+    struct sol_lwm2m_payload payload)
+{
+    int r;
+
+    if (client->is_bootstrapping) {
+        clear_bootstrap_ctx(client);
```

Comment:
You forgot to call `clear_bootstrap_ctx` when create/read is called/execute. I suggest that you move the logic above to handle_resource